### ISSUE TITLE
Request for versioned docs

### ISSUE THREAD

**Issue by eddgrant** [hashicorp/terraform#12941]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
0.8.8

### Affected Resource(s)
Documentation

### Terraform Configuration Files
N/A

### Debug Output
N/A

### Panic Output
N/A

### Expected Behavior
As a user of a specific version of Terraform I should be able to access the documentation relating to that specific Terraform version 

### Actual Behavior
The online documentation seems to only reflect the current latest version

### Steps to Reproduce
1. Open the online documentation e.g.: https://www.terraform.io/docs/providers/index.html
2. Try and find documentation relating to functionality which exists in the specific version of Terraform (which isn't the latest)

### Important Factoids
This is particularly relevant given the recent release of 0.9.0 and the fact that so many people use Terragrunt. Many folks may not be ready to make the switch to 0.9.0 and ditch Terragrunt yet. Getting rid of the 0.8.x docs is problematic as the 0.9.x docs now don't contain any reference to some functions which I presume were removed in 0.9 (for example the commands to manage remote state in <= 0.9.0 versions).

Would it be possible to get the docs versioned? It would probably be a useful thing to have going forward too.

### References
N/A

**Comment by sysadmiral** [comment]
If versioned docs is not an option perhaps under a data_source's or resource's documentation add an `introduced_in_version` or a `deprecated_in_version` field maybe?

**Comment by apparentlymart** [comment]
Hi @eddgrant, @sysadmiral! Sorry for the silence here.

Versioned docs is definitely something we intend to do. There are some complexities to figure out with how we generate the website today, so it'll take some work to get there, but we understand the need and it's definitely a goal, and exacerbated by the fact that after 0.10 providers and core will be on separate release cycles.

We'll have some more on this once we get a plan on how to evolve our website generation process to support this.

**Comment by eddgrant** [comment]
Hey @apparentlymart , thanks for the head up. Look forward to the new docs when they arrive.

**Comment by zenbiking** [comment]
Any updates on this?

**Comment by kkrusedp** [comment]
Hello,

Squeak Squeak, Oiling the squeaky wheel here.  Any updates on this?

**Comment by chris** [comment]
Yes +1 for this, key need, as existing docs have options and examples that often don't work in older versions.

**Comment by joewilliams** [comment]
Another +1 for this, we are running into this quite a bit since we don't run the latest.

**Comment by sarcasticadmin** [comment]
As called out in https://github.com/hashicorp/consul/issues/4964 `react-native` has a nice solution to this: https://facebook.github.io/react-native/versions

Just having a reference to these doc versions on completely dedicated page would be nice. It could even be similar to the `You can also download older versions of Terraform from the releases service` 
 link found on the terraform downloads page that allows you to get any version of the binary: https://www.terraform.io/downloads.html

**Comment by anthony-coder** [comment]
+1 Bumping this issue. Would be useful to have the ability to navigate previous documentation versions.

**Comment by markbennettcanada** [comment]
Another bump on this. Would be extremely helpful.

**Comment by mildwonkey** [comment]
Please do not post "+1" comments here, since it creates noise for others watching the issue and ultimately doesn't influence our prioritization because we can't actually report on these. Instead, react to the original issue comment with 👍, which we [can and do report on](https://github.com/hashicorp/terraform/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) during prioritization.

**Comment by Murty0** [comment]
I just found this: https://github.com/terraform-providers/terraform-provider-aws/tree/v2.35.0/website/docs/r

**Comment by paultyng** [comment]
Versioned docs for providers is now live in the registry, we will start redirecting docs pages to these in the future: https://www.hashicorp.com/blog/announcing-providers-in-the-new-terraform-registry
